Update Readme for Traditional NLP experiments/ (#95)
Princeton-CDH/remarx: README.md

Reviewer: Is it worth documenting at a high-level what experiments were done for each task? Right now there's nothing tied to quotation detection.

traditional NLP approaches and out-of-the-box software to identify
titles and quotations in sample text content from the Citing Marx project.

### Preliminary conclusions

- lemmatized search is decent, but doesn't find variants with OCR errors, typos, hyphenation
- Flair worked better than Stanza for our use cases; Flair NER does identify some instances of titles with OCR errors ("Eanifests", "Mauifests") but does not find all titles

>   - Available NER models for German only have 4 categories (PER, LOC, ORG, MISC) compared to the richer 18-category ontology available for English

Reviewer: ```suggestion
  - Available off-the-shelf NER models for German only have 4 categories (PER, LOC, ORG, MISC) compared to the richer 18-category ontology available for English.
  - Fine-grained NER annotation datasets exist for German but these may not include a relevant category (e.g., work of art) and likely require building our own NER pipeline.
```

### Contents

> - `data/`

Reviewer: What about the seq2seq directory?

Author: whoops! thanks for catching

  - `annotation/`: snapshot of annotation data exported from Recogito; CSV files
  - `direct_quotes_subset.csv`: subset of direct quote annotations from experiment 1
  - `title_mentions_subset.csv`: subset of title mention annotations from experiment 1
  - `compiled_title_mentions.csv`: restructured title annotation data generated by `compile_title_annotations.py` script with Yes/No/Maybe values for mentions of Kapital and Manifesto
  - `title_searchphrases.csv` : CSV file with search phrases and corresponding titles

>   - `candidate-mentions/` : files generated by `remarx-candidate-mentions` lemmatized search

Reviewer: Did you make a command for `candidate_mentions.py`? If so, this should be documented.

    - `title_mentions.csv`: results from several title phrases run across three text files; refer to `title_searchphrases.csv` for mapping phrases to canonical titles
    - `concept_mentions.csv`: results from a few concept phrases run against all 25 _Die Neue Zeit_ texts

>    - `candidate_ner.csv` : CSV file with spans tagged by NER with Flair, as possible candidates for title mentions; one entity per row

Reviewer: ```suggestion
   - `candidate_ner.csv` : CSV file with spans tagged by NER with Flair generated by `get_ner_spans.py` script for the sentences in `title_mentions.csv`; one NER entity per row
```

>    - `title_mentions_flair_ner.csv`: CSV file with preliminary Flair NER results; multiple entities by type for each sentence id

Reviewer: This was produced by one of the notebooks, right?

>    - `title_mentions_sent_results.csv`: CSV combining compiled title mentions data (with yes/no/maybe for each title), lemmatized search results for title candidates, and Flair NER spans

Reviewer: Might be worth documenting how it was generated... It was produced using the methods in polars_utils.py within one of the notebooks (not sure if that logic made it into the final notebook)

- `notebooks/`: marimo notebooks for data exploration, experimentation, and review

> - `src/remarx/`: python package with utility code and scripts

Reviewer: Missing `compile_title_annotations.py` script. Brief documentation should be added.

   - `build_sentence_corpus.py`: script to parse a directory of text files into a JSONL corpus of sentences; uses Stanza for sentence splitting
   - `candidate_mentions.py`: script to run lemmatized search on directory of text files and generate a CSV of matching sentences
   - `get_ner_spans.py`: script to run Flair NER on a CSV of candidate sentences and generate a CSV of identified entities, one entity per row

>    - `polar_utils.py`: utility method sfor loading and manipulating data related to title mentions and candidates

Reviewer: ```suggestion
   - `polar_utils.py`: utility methods for loading and manipulating data related to title mentions and candidates
```

  - `compiled_title_mentions.csv`: restructured title annotation data generated by `compile_title_annotations.py` script with Yes/No/Maybe values for mentions of Kapital and Manifesto
  - `title_searchphrases.csv` : CSV file with search phrases and corresponding titles
  - [`candidate-mentions/`](data/candidate_mentions) : files generated by and documentation related to `candidate_mentions.py`  lemmatized search; see the README for this folder
    - `title_mentions.csv`: results from several title phrases run across three text files; refer to `title_searchphrases.csv` for mapping phrases to canonical titles
    - `concept_mentions.csv`: results from a few concept phrases run against all 25 _Die Neue Zeit_ texts
    - `candidate-mentions.slurm` sample slurm batch file for running this command; more details in the folder README file
   - `candidate_ner.csv` : CSV file with spans tagged by NER with Flair generated by `get_ner_spans.py` script for the sentences in `title_mentions.csv`; one NER entity per row
   - `title_mentions_flair_ner.csv`: CSV file with preliminary Flair NER results; multiple entities by type for each sentence id

>    - `title_mentions_sent_results.csv`: CSV combining compiled title mentions data (with yes/no/maybe for each title), lemmatized search results for title candidates, and Flair NER spans. Genearted with methods in `remarx.polars_utils` and `evaluate-candidate-sentences` notebook

Reviewer: ```suggestion
   - `title_mentions_sent_results.csv`: CSV combining compiled title mentions data (with yes/no/maybe for each title), lemmatized search results for title candidates, and Flair NER spans. Generated with methods in `remarx.polars_utils` and `evaluate-candidate-sentences` notebook
```
